### ISSUE TITLE
QtWebEngine Distro Support

### ISSUE THREAD

**Issue by danieleds** [notepadqq/notepadqq#674]
Just to recap and keep track of the current problems with shipping Notepadqq with QtWebEngine:

 * In Ubuntu, qtwebengine is packaged only since 17.10.
 * Snap uses Ubuntu Xenial 16.04 by default, which doesn't ship qtwebengine.
   - "core18" can be used to base the snap on Ubuntu 18.04, however it is not supported yet on Canonical's build system, and there is no ETA afaik.

**Comment by JuBan1** [comment]
So at the moment we're left with not being able to ship to anything below 17.10 *and* not being able to ship snaps at all?

**Comment by danieleds** [comment]
Kind of. There is the possibility of building Qt 5.8 from source in the snap, but it probably means 2 days of build time.

We can also build core18 snaps *locally* (w/o the automated build system) and then upload them to the store. Not ideal either.

Anyway, we have time to think about this. I don't want to ship the current code to the general public until we're totally sure that everything works, an this will take weeks.

**Comment by hatdropper1977** [comment]
I can't seem to find any repo on Centos 7 that provides **qt-webengine-devel**.

```bash
sudo yum list qt5-qtweb*
Loaded plugins: fastestmirror, langpacks
Loading mirror speeds from cached hostfile
 * base: linux.cc.lehigh.edu
 * epel: mirror.cogentco.com
 * extras: mirror.wdc1.us.leaseweb.net
 * updates: repos-va.psychz.net
Installed Packages
qt5-qtwebchannel.x86_64	 			5.9.2-1.el7		@base
qt5-qtwebkit.x86_64					5.9.1-1.el7		@epel
qt5-qtwebkit-devel.x86_64				5.9.1-1.el7		@epel
qt5-qtwebsockets.x86_64				5.9.2-1.el7		@base
qt5-qtwebsockets-devel.x86_64			5.9.2-1.el7		@base
Available Packages
qt5-qtwebchannel.i686					5.9.2-1.el7		base 
qt5-qtwebchannel-devel.i686			        5.9.2-1.el7		base 
qt5-qtwebchannel-devel.x86_64			5.9.2-1.el7		base 
qt5-qtwebchannel-doc.noarch			        5.9.2-1.el7		base 
qt5-qtwebchannel-examples.x86_64	        5.9.2-1.el7		base 
qt5-qtwebsockets.i686					5.9.2-1.el7		base 
qt5-qtwebsockets-devel.i686			        5.9.2-1.el7		base 
qt5-qtwebsockets-doc.noarch			        5.9.2-1.el7		base 
qt5-qtwebsockets-examples.x86_64		5.9.2-1.el7		base 
```

As a result I can't build v1.4.0.

My configure fails with:
```bash
$ ./configure --prefix /usr | grep 'not found'
checking for Qt5WebEngine library... not found!
checking for Qt5WebEngineWidgets library... not found!
```
Then a **provides** command yields nothing:
```bash
$ sudo yum provides Qt5WebEngine
Loaded plugins: fastestmirror, langpacks
Loading mirror speeds from cached hostfile
 * base: linux.cc.lehigh.edu
 * epel: mirror.cogentco.com
 * extras: mirror.wdc1.us.leaseweb.net
 * updates: repos-va.psychz.net
No matches found
```
I can succesfully build an install **v1.3.6** but that has the 'right click reload' #336 bug on my system.

Anybody have any helpful hints on how to get the **qt-webengine-devel** package on Centos 7?

**Comment by danieleds** [comment]
@hatdropper1977 [v1.4.8](https://github.com/notepadqq/notepadqq/tree/v1.4.8) still use QtWebKit and has the bug fixed, you should be able to build it.

**Comment by hatdropper1977** [comment]
That fixed it, thanks!

**Comment by darealshinji** [comment]
Does snapcraft support PPAs? You could use a PPA with pre-built newer Qt stuff then. Or maybe you use the pre-built libraries from the installer provided by qt.io.

**Comment by danieleds** [comment]
@darealshinji We can add a PPA, but I didn't manage to understand how to make it work with the snapcraft plugins (qmake, desktop-qt5). I didn't put much time into it though.

**Comment by JuBan1** [comment]
@danieleds Could you tell me how you build the core18 snap? I've been trying to get snapcraft to work for debugging snaps but I can't it to work properly.

**Comment by darealshinji** [comment]
@JuBan1 add the line `base: core18` like here: https://github.com/notepadqq/notepadqq/commit/9006f8bcc4d5f196e7b87825632dbd8b6ba00a6e

**Comment by scx** [comment]
@hatdropper1977 
> I can't seem to find any repo on Centos 7 that provides qt-webengine-devel.

I have already created a COPR repository for EPEL7: https://github.com/notepadqq/notepadqq/issues/668#issuecomment-401185470

**Comment by danieleds** [comment]
This is fixed with 52f09316c450cad138921e3ca2e848885fcc3ee8, 27588db456216d7fd41cc1170ad43bfbde1dc8a4, 0076368408ef1260089aef6d898d12bebafb4e2a.

We're now able to ship working snaps with core18, even if the build process cannot be automated yet (canonical's build server still only support core16).

Classic distro packages for new nqq versions should be able to use qtwebengine. Old ubuntu version can:

 - stay on nqq 1.4.8, or
 - use the snap package, or
 - build from source
